lib/marc_match_key/key.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+### A collection of functions and classes to generate a match key based on the
+###   GoldRush algorithm
+module MarcMatchKey
+  ### Generates the Gov Doc portion of a GoldRush key
+  class Key
+    attr_reader :record, :key
+
+    def initialize(record)
+      @record = record
+      @key ||= generate_key
+    end
+
+    private
+
+    def generate_key
+      match_key = key_part_one
+      match_key << key_part_two
+      match_key << key_part_three
```

Comment:
To my eye, using string interpolation here would make it clearer that it returns a string.
```suggestion
      "#{key_part_one}#{key_part_two}#{key_part_three}"
```

Reply:
I thought an array might be easier to work with if there need to be more parts in the future.